geps/gep-1762/index.md

```diff
@@ -59,6 +59,9 @@ With this configuration, an implementation:
 
 * MUST mark the Gateway as `Programmed` and provide an address in `Status.Addresses` where the Gateway can be reached on each configured port.
 * MUST label all generated resources (Service, Deployment, etc) with `gateway.networking.k8s.io/gateway-name: my-gateway` (where `my-gateway` is the name of the Gateway resource).
+  * In the case that a Gateway resource name is longer than the maximum label value length of 63 characters, implementations MUST set the value of the label to the first 20 characters of the Gateway name with a `-` character appended and then the `sha1` checksum of the remainder of the name (40 characters long) appended.
```

Comment:
Why not just use Kubernetes `generateName` field? If we need to predictably tie this back to a Gateway, we already have the label defined above. I'm guessing we want to avoid stomping over any preexisting resources with generated ones, and anything too prescriptive here would effectively _require_ that behavior. On the other hand, `generateName` ensures that we always get a unique + new name and we can rely on label selection for reliable lookups instead of name.

Reply:
hm, not sure how the `generateName` field helps here? This update isn't for specifying the name of generated resources but what to do with the `gateway.networking.k8s.io/gateway-name` label value (which should be set to the Gateway name) in the case a Gateway's name is > 63 characters

Reply:
🤦 I definitely did not read this clearly enough, sorry for the confusion!